File: apps/sepuling-java/src/main/java/smecalculus/bezmen/storage/SepulkaDao.java
package smecalculus.bezmen.storage;

import java.util.Optional;
import java.util.UUID;
import smecalculus.bezmen.core.StateDm;
import smecalculus.bezmen.core.StateDm.AggregateRoot;
import smecalculus.bezmen.core.StateDm.Existence;
import smecalculus.bezmen.core.StateDm.Preview;
import smecalculus.bezmen.core.StateDm.Touch;

/**
 * Port: server side
 */
public interface SepulkaDao {
    StateDm.AggregateRoot add(AggregateRoot state);

    Optional<Existence> getBy(String externalId);

    Optional<Preview> getBy(UUID internalId);

    void updateBy(Touch state, UUID internalId);
}

File: apps/sepuling-kotlin/src/main/kotlin/smecalculus/bezmen/storage/SepulkaDao.kt
package smecalculus.bezmen.storage

import smecalculus.bezmen.core.StateDm
import java.util.UUID

/**
 * Port: server side
 */
interface SepulkaDao {
    fun add(state: StateDm.AggregateRoot): StateDm.AggregateRoot

    fun getBy(externalId: String): StateDm.Existence?

    fun getBy(internalId: UUID): StateDm.Preview?

    fun updateBy(
        state: StateDm.Touch,
        internalId: UUID,
    )
}
